lib/rdoc/generator/darkfish.rb

```diff
@@ -729,6 +729,10 @@ def excerpt(comment)
     extracted_text[0...150].tr_s("\n", " ").squeeze(" ")
   end
 
+  def canonical_url(context)
+    File.join(@options.canonical_root, context&.path.to_s)
```

Comment:
I prefer making `canonical_url` taking the path already instead of a nilable context. It'll likely be easier to maintain and debug.

Reply:
I've changed the implementation to take the path instead.